src/include/processor/operator/result_collector.h

```diff
@@ -28,6 +28,8 @@ class FTableSharedState {
 
     inline std::shared_ptr<FactorizedTable> getTable() { return table; }
 
+    inline void setTable(std::shared_ptr<FactorizedTable> other) { table = other; }
```

Comment:
hmmm if you wanna add this setter. Open an issue and assign to me saying we should refactor FTableSharedState to always construct FTable during mapping